src/components/Content.jsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react";
+import { loadData } from "./loadData";
+
+const Content = () => {
+  const [posts, setPosts] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    loadData(setPosts, setError);
```

Comment:
Instead of passing `setProps` and `setError` as arguments, what do you think about using them here directly into the promise `loadData` returns? This way we reduce the API surface of the `loadData` function, making it less prone to bugs.